python/pdo/contract/replication.py

```diff
@@ -22,18 +22,15 @@
 import logging
 logger = logging.getLogger(__name__)
 
-# we many want to parametrize this later
-num_threads_per_storage_service = 2
-
 # -----------------------------------------------------------------
-__replication_manager_executor__ = ThreadPoolExecutor(max_workers=1)
-__pending_replication_tasks_manager_queue__ = queue.Queue()
+replication_threads = []
+replication_thread_lock = threading.Lock()
```

Comment:
queue.Queue() is thread safe AFAIK, "The [Queue](https://docs.python.org/3/library/queue.html#queue.Queue) class in this module implements all the required locking semantics."